provider/infoblox/infoblox_test.go

```diff
@@ -393,50 +386,112 @@ func newInfobloxProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provid
 	}
 }
 
+// TestInfobloxRecords function checks if the current records
+// are fetched properly by the ProviderConfig.Records method.
 func TestInfobloxRecords(t *testing.T) {
-	client := mockIBConnector{
-		mockInfobloxZones: &[]ibclient.ZoneAuth{
-			createMockInfobloxZone("example.com"),
-			createMockInfobloxZone("other.com"),
+	testCases := map[string]struct {
+		client            mockIBConnector
+		domainFilter      endpoint.DomainFilter
+		createPtr         bool
+		expectedEndpoints []*endpoint.Endpoint
+	}{
+		// A couple of CNAME/A records match the domain filter. Create PTR is disabled.
+		"CNAMEAndARecordsMatchDomainFilter": {
```

Comment:
I can see a deduplication handling but not domain filtering mentioned in the testcase name. Can you please add some DNS records which don't match the specified domain so that we can see the client side domain filtering?